x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_detail.tsx

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useMemo, useEffect, Fragment } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiI18nNumber,
+  EuiSpacer,
+  EuiText,
+  EuiDescriptionList,
+  EuiCode,
+  EuiTextColor,
+  EuiTitle,
+} from '@elastic/eui';
+import styled from 'styled-components';
+import { useSelector } from 'react-redux';
+import { CrumbInfo, formatDate, StyledBreadcrumbs } from './panel_content_utilities';
+import * as event from '../../../../common/endpoint/models/event';
+import { ResolverEvent } from '../../../../common/endpoint/types';
+import * as selectors from '../../store/selectors';
+import { useResolverDispatch } from '../use_resolver_dispatch';
+import { PanelContentError } from './panel_content_error';
+
+export const BoldCode = styled(EuiCode)`
+  &.euiCodeBlock code.euiCodeBlock__code {
+    font-weight: 900;
+  }
+`;
+
+/**
+ * A helper function to turn objects into EuiDescriptionList entries.
+ * This reflects the strategy of more or less "dumping" metadata for related processes
+ * in description lists with little/no 'prettification'. This has the obvious drawback of
+ * data perhaps appearing inscrutable/daunting, but the benefit of presenting these fields
+ * to the user "as they occur" in ECS, which may help them with e.g. EQL queries.
+ *
+ * @param {object} obj The object to turn into `<dt><dd>` entries
+ */
+const objectToDescriptionListEntries = function* (
+  obj: object,
+  prefix = ''
+): Generator<{ title: string; description: string }> {
+  const nextPrefix = prefix.length ? `${prefix}/` : '';
+  for (const [metaKey, metaValue] of Object.entries(obj)) {

```

Comment:
Is Object.entries(obj) recreated on every iteration of the loop here?

Reply:
Good question - The way I _thought_ it works was that `for...of` only runs the iterator once, but I'm gonna check and make sure.

Reply:
`````js
var myIterable = {}
let callCount = 0;
myIterable[Symbol.iterator] = function* () {
    callCount++;
    yield 1;
    yield 2;
    yield 3;
};
for(i of myIterable){
    console.info(i);
}
console.assert(callCount === 1);
`````
I think that means it only goes once, right?

Reply:
Yep!